test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -0,0 +1,208 @@
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair�s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair�s trademarks, including but not limited to "PBS�",
+# "PBS Professional�", and "PBS Pro�" and Altair�s logos is subject to Altair's
+# trademark licensing policies.
+
+import os
+import ntpath
+
+from tests.functional import *
+from ptl.utils.pbs_snaputils import *
+
+
+class TestPBSSnapshot(TestFunctional):
+    """
+    Test suit with unit tests for the pbs_snapshot tool
+    """
+
+    def test_usage_message(self):
+        """
+        Test whether the '-h' option to pbs_snapshot works
+        """
+        res = self.du.run_cmd(cmd=["pbs_snapshot", "-h"])
+        self.assertTrue(res['rc'] == 0)
+
+    def test_capture_server(self):
+        """
+        Test the 'capture_server' interface of PBSSnapUtils
+        """
+
+        # Set something on the server so we can match it later
+        job_hist_duration = "12:00:00"
+        attr_list = {"job_history_enable": "True",
+                     "job_history_duration": job_hist_duration}
+        self.server.manager(MGR_CMD_SET, SERVER, attr_list)
+
+        output_dir = os.path.join("/", "tmp", "snapshot_server")
+        snap_args = {"out_dir": output_dir, "num_service_logs": 2,
+                     "num_acct_logs": 5}
+
+        with PBSSnapUtils(snap_args) as snap_obj:
+            snap_obj.capture_server(True, True)
+
+        # Go through the snapshot and perform certain checks
+        # Check 1: the snapshot exists
+        self.assertTrue(os.path.isdir(output_dir))
+        # Check 2: all directories except the 'server' directory have no files
+        outdir_fullpath = os.path.abspath(output_dir)
+        svr_fullpath = os.path.join(outdir_fullpath, "server")
+        for root, _, files in os.walk(output_dir):
+            for file in files:
+                file_fullpath = os.path.join(root, file)
+                # Find the common paths between 'server' & the file
+                common_path = os.path.commonprefix([file_fullpath,
+                                                    svr_fullpath])
+                self.assertTrue(ntpath.basename(common_path) == "server")
+        # Check 3: qstat_Bf.out exists
+        qstat_bf_path = os.path.join(output_dir, "server", "qstat_Bf.out")
```

Comment:
Can't you dig out the 'qstat_Bf.out' name from PBSSnapUtils() itself?  That way you don't have to worry about changing this test if you ever change the name of the file.